framework/include/utils/PiecewiseByBlockLambdaFunctor.h

```diff
@@ -261,6 +261,14 @@ PiecewiseByBlockLambdaFunctor<T>::evaluate(const Moose::FaceArg & face,
 {
   using namespace Moose::FV;
   mooseAssert(state == 0, "Only current time state supported.");
+
+  if (face.fi->boundaryIDs().size())
+  {
+    Moose::SingleSidedFaceArg ssfa = {
+        face.fi, face.limiter_type, face.elem_is_upwind, face.correct_skewness, face.elem_sub_id};
+    return this->evaluate(ssfa, 0);
+  }
+
```

Comment:
I'm not a fan of this. You can have internal sidesets that may be totally unrelated to the caller of this evaluation

Reply:
I added a different condition which is more robust. By the way it is amazing how every comment of this preliminary PR pointed to things that actually caused issues during the testing phase.